### ISSUE TITLE
Feature Request: File-based configuration for CircuitBreakerRegistry

### ISSUE THREAD

**Issue by brandonheck** [resilience4j/resilience4j#957]
Resilience4j version: resilience4j-circuitbreaker-1.3.1

Java version: 8

Is there a standard way to configure resilience4j's circuit breaker, similar to what is in [resilience4j-spring](https://github.com/resilience4j/resilience4j/tree/master/resilience4j-spring/) or [resilience4j-ratpack](https://github.com/resilience4j/resilience4j/tree/master/resilience4j-ratpack), but generically using a file rather than tying to a web container's configuration mechanism.

**Comment by RobWin** [comment]
No, there is no existing module, but you could create one using Apache Commons Configuration.

**Comment by brandonheck** [comment]
I'm happy to take a swing at a PR for this functionality, but I'd like to get some guidelines on approach. Would the preference be to have a separate module that can be used for reading configuration for any of the `resilience4j` core modules, or would it be preferable to add the `commons-configuration` directly to `resilience4j-circuitbreaker`? FWIW, I would also be interested in contributing a Bulkhead configuration.

**Comment by RobWin** [comment]
It should be a separate module.
